### PR TITLE
add Apache APISIX

### PR DESCRIPTION
**Pull request by tao12345666333**
[Apache APISIX](https://apisix.apache.org/) is a dynamic, real-time, high-performance API gateway.

[APISIX](https://github.com/apache/apisix/) provides rich traffic management features such as load balancing, dynamic upstream, canary release, circuit breaking, authentication, observability, and more.

You can use Apache APISIX to handle traditional north-south traffic, as well as east-west traffic between services. It can also be used as a [k8s ingress controller](https://github.com/apache/apisix-ingress-controller/).

### CODE REVIEW

File: packages/apisix/apisix.patch [hunk 1]
```diff
@@ -0,0 +1,10 @@
+diff -x '*.tgz' -x '*.lock' -uNr packages/apisix/charts-original/Chart.yaml packages/apisix/charts/Chart.yaml
+--- packages/apisix/charts-original/Chart.yaml
++++ packages/apisix/charts/Chart.yaml
+@@ -20,3 +20,6 @@
+ name: apisix
+ type: application
+ version: 0.3.3
++annotations:
++  catalog.cattle.io/certified: partner
++  catalog.cattle.io/release-name: apisix
```
Comment (PennyScissors) [inline packages/apisix/apisix.patch:10]: Could you please add this additional annotation for the name that will be displayed in the UI?
catalog.cattle.io/display-name: Apache APISIX
I used Apache APISIX as an example, but you can change it to any other name you think is best.

**Review by PennyScissors — CHANGES_REQUESTED** [review]
Could you please take me through the steps you took to deploy this chart in Rancher for testing? I tried deploying it in Rancher v2.5.8, but it times out and fails.

File: packages/apisix/apisix.patch (hunk 1 above)
Comment (tao12345666333) [reply packages/apisix/apisix.patch:10]: Thanks! I agree that Apache APISIX is better.

**Comment by tao12345666333** [comment]
@PennyScissors thanks for review! 

>but it times out and fails.

can you try to run `curl -I  https://charts.apiseven.com/apisix-0.3.3.tgz`

this is my output:

```
$ curl -I  https://charts.apiseven.com/apisix-0.3.3.tgz
HTTP/2 200 
content-type: application/octet-stream
content-length: 51933
date: Sat, 12 Jun 2021 09:45:00 GMT
last-modified: Thu, 10 Jun 2021 11:20:55 GMT
etag: "60c1f597-cadd"
accept-ranges: bytes
server: APISIX/2.3

```

The current chart repo is deployed on a host in Azure China.

**Comment by PennyScissors** [comment]
> @PennyScissors thanks for review!
> 
> > but it times out and fails.
> 
> can you try to run `curl -I https://charts.apiseven.com/apisix-0.3.3.tgz`
> 
> this is my output:
> 
> ```
> $ curl -I  https://charts.apiseven.com/apisix-0.3.3.tgz
> HTTP/2 200 
> content-type: application/octet-stream
> content-length: 51933
> date: Sat, 12 Jun 2021 09:45:00 GMT
> last-modified: Thu, 10 Jun 2021 11:20:55 GMT
> etag: "60c1f597-cadd"
> accept-ranges: bytes
> server: APISIX/2.3
> ```
> 
> The current chart repo is deployed on a host in Azure China.

I get the same output.

**Comment by PennyScissors** [comment]
This is what I see in my Rancher v2.5.8 setup when I try to install it. It continues like this until it times out.
image

**Comment by tao12345666333** [comment]
Thank you for the information, let me check!

**Comment by PennyScissors** [comment]
Hello @tao12345666333, we recently made some changes in the core scripts of this repository that changes the workflow to add a chart, and would require you to refactor this PR to the point that it's better to just submit a new one. Please check the updated README.md in the main-source branch, and submit a new PR whenever you are ready. Thanks!

**Comment by tao12345666333** [comment]
Thanks for mention @PennyScissors

**Pull request closed without merging**